rust/src/nasl/builtin/network/socket.rs

```diff
@@ -618,161 +523,250 @@ impl NaslSockets {
         ))
     }
 
-    fn open_sock_tcp_ip(
+    /// Reads the information necessary for a TLS connection from the KB and
+    /// return a TlsConfig on success.
+    fn get_tls_conf(context: &Context) -> Result<TlsConfig, FunctionErrorKind> {
+        let cert_path = match get_kb_item(context, "Secret/tls_cert")? {
+            Some(x) => Ok(x.to_string()),
+            None => Err(FunctionErrorKind::Diagnostic(
+                "KB key 'Secret/tls_cert' is not set".to_string(),
+                None,
+            )),
+        }?;
+
+        let key_path = match get_kb_item(context, "Secret/tls_key")? {
+            Some(x) => Ok(x.to_string()),
+            None => Err(FunctionErrorKind::Diagnostic(
+                "KB key 'Secret/tls_key' is not set".to_string(),
+                None,
+            )),
+        }?;
+
+        let password = match get_kb_item(context, "Secret/tls_password")? {
+            Some(x) => Ok(x.to_string()),
+            None => Err(FunctionErrorKind::Diagnostic(
+                "KB key 'Secret/tls_password' is not set".to_string(),
+                None,
+            )),
+        }?;
+
+        let cafile_path = match get_kb_item(context, "Secret/tls_cafile")? {
+            Some(x) => Ok(x.to_string()),
+            None => Err(FunctionErrorKind::Diagnostic(
+                "KB key 'Secret/tls_cafile' is not set".to_string(),
+                None,
+            )),
+        }?;
```

Comment:
```suggestion rust
        let get_kb_string = |name| {
            match get_kb_item(context, name)? {
                    Some(x) => Ok(x.to_string()),
                    None => Err(FunctionErrorKind::Diagnostic(
                        format!("KB key {} is not set", name),
                        None,
                    )),
                }
        };

        let cert_path = get_kb_string("Secret/tls_cert")?;
        let key_path = get_kb_string("Secret/tls_key")?;
        let password = get_kb_string("Secret/tls_password")?;
        let cafile_path = get_kb_string("Secret/tls_cafile")?;
```

Reply:
or better yet make the closure a function and use it in `open_sock_kdc` as well

Reply:
or better yet move the closure as a function besides get_kb_item and call it  kb_item_err or similar. That way this case would be available for everybody.